tokio/src/io/util/async_read_ext.rs

```diff
@@ -180,9 +182,16 @@ cfg_io_util! {
         ///
         /// # Return
         ///
-        /// On a successful read, the number of read bytes is returned. If the
-        /// supplied buffer is not empty and the function returns `Ok(0)` then
-        /// the source has reached an "end-of-file" event.
+        /// If the return value of this method is `Ok(n)`, then it must be
+        /// guaranteed that `0 <= n <= buf.len()`. A nonzero `n` value indicates
```

Comment:
The [`BufMut`][1] trait has no `len` method.

[1]: https://docs.rs/bytes/1/bytes/trait.BufMut.html

Reply:
Ah, you're right. I'll update.